src/Microsoft.Diagnostics.Runtime/Native/NativeHeap.cs

```diff
@@ -23,6 +23,76 @@ internal class NativeHeap : HeapBase
         private ISymbolProvider _symProvider;
         private Dictionary<NativeModule, ISymbolResolver> _resolvers = new Dictionary<NativeModule, ISymbolResolver>();
 
+        internal class NativeStackFrame : ClrStackFrame
+        {
+            private Address _ip;
+            private string _symbolName;
+            private ClrModule _module;
+
+            public NativeStackFrame(Address ip, string symbolName, ClrModule module)
+            {
+                _ip = ip;
+                _symbolName = symbolName;
+                _module = module;
+            }
+
+            public override string DisplayString
+            {
+                get
+                {
+                    return $"{_symbolName}";
+                }
+            }
+
+            public ClrModule Module
+            {
+                get
+                {
+                    return _module;
+                }
+            }
+
+            public override ulong InstructionPointer
+            {
+                get
+                {
+                    return _ip;
+                }
+            }
+
+            public override ClrStackFrameType Kind
+            {
+                get
+                {
+                    throw new NotImplementedException("This information is unavailable when debugging Native code");
```

Comment:
As mentioned above, the design of ClrMD is to not throw exceptions except in unrecoverable cases.  Can you add a new entry to ClrStackFrameType called Unknown, the `return ClrStackFrameType.Unknown;` here?

    public enum ClrStackFrameType
    {
        /// <summary>
        /// Indicates that information on this stack frame type is unavailable
        /// </summary>
        Unknown = -1,

        /// <summary>
        /// Indicates this stack frame is a standard managed method.
        /// </summary>
        ManagedMethod = 0,

        /// <summary>
        /// Indicates this stack frame is a special stack marker that the Clr runtime leaves on the stack.
        /// Note that the ClrStackFrame may still have a ClrMethod associated with the marker.
        /// </summary>
        Runtime = 1
    }